node/src/components/small_network.rs

```diff
@@ -330,12 +331,28 @@ where
 
         let chain_info = chain_info_source.into();
         let protocol_version = chain_info.protocol_version;
+
+        let keylog = match cfg.keylog_path {
```

Comment:
Since the binding is `Option<_>` anyway, I'd suggest using a `.map()`

Reply:
My bad, good catch!

Reply:
Ah, I was a little hasy perhaps - there's a `?` operator returning an error. Using `map` would then make the code require something like `.map(|path| { ... Ok(writer)}).transpose()?` to thread through the error and require a type annotation. Additionally, `cfg.keylog_path.as_ref()` would have to be used to avoid the move.

I think at that point the explicit version becomes preferable for sheer readability.